nulib/meadow
# Integrate `?as=iiif` Manifests from DC API

# Summary 

Meadow now uses DCAPI `as=iiif` manifests rather than it's own internally generated manifests.

# Specific Changes in this PR

 - GraphQL `work` query returns updated manifestUrl
 - `getIiifManifestHeaders` GQL query removed
-  expose `dcapiEndpoint` query in GQL
- Remove IIIF manifest generation & updating code
    - Existing manifests still exist in S3 and are available at existing url's but no further updating will be performed
- Work Screen / Viewer instances 
   - utilizes Clover IIIF Viewer for all work types and Openseadragon React Viewer is removed
   - poll graphql for DC API bearer tokens every 300 seconds
   - request an `etag` header every 10 seconds
   - `etag` used as key in Clover instances

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test

- In dev make sure to install new assets and dependencies
  - Create some new audio + video works (existing ones may not have transcoded the AV assets - this was a recent Meadow change, you can verify this as manifests would have .m3u8 uri's and you'll have transcoded assets in your streaming bucket rather than the original uploaded file)
- In dev, start your DCAPI through the `https-proxy`
- Verify on work pages (all work types) should now get IIIF manifests from DC API and display in Clover viewer
- Verify etag polling (updates like file set label edits, re-ordering or file set additions/deletions should be reflected in UI/Clover) 
- Verify nothing else is broken
- Verify updates to existing s3 manifests have stopped but existing manifests still available at old url

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes


# Tested/Verified
